### ISSUE TITLE
SolrNet doesn't re-use http connections when querying solr

### ISSUE THREAD

**Issue by CiprianMo** [SolrNet/SolrNet#411]
Hi we noticed that SolrNet opens new http connections for every query. We are using StructureMap.SolrNetIntegration, and it seems that the problem is the SolrNetRegistry. When the ISolrConnection is registered with AutoSolrConnection it is not registered as a Singleton. So whenever we are querying a core a new connection is created. 
Do you agree that this a problem ?

**Comment by gjunge** [comment]
@CiprianMo  Yes it is indeed better to have it declared as a singleton. We already do the same in the `Microsoft.DependencyInjection.SolrNet` package. 

Do you want to create a  PR for this?

**Comment by rajupaladiya** [comment]
Hello Team,
Any update on this ?
This is creating issue on performance while site have around 100 online same user.

**Comment by gjunge** [comment]
Please feel free to create a PR.
